src/Sylius/Bundle/CoreBundle/Doctrine/ORM/PaymentMethodRepository.php

```diff
@@ -26,4 +28,18 @@ public function createListQueryBuilder()
             ->leftJoin('o.translations', 'translation')
         ;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findEnabledForChannel(ChannelInterface $channel)
```

Comment:
Then we would not need this custom repository, we can use it directly in the resolver or implement on the model - see http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/working-with-associations.html#filtering-collections.